File: languages/go/exercises_solutions/quick_sort_soln.go
// func quickSort(arr []int) []int {
// 	if len(arr) <= 1 {
// 		return arr
// 	}

// 	pivot := arr[0]
// 	left := []int{}
// 	right := []int{}

// 	for _, val := range arr[1:] {
// 		if val >= pivot {
// 			right = append(right, val)
// 		} else {
// 			left = append(left, val)
// 		}
// 	}
// 	answer := []int{}
// 	sortedLeft := quickSort(left)
// 	sortedRight := quickSort(right)
// 	answer = append(answer, sortedLeft...)
// 	answer = append(answer, pivot)
// 	answer = append(answer, sortedRight...)
// 	return answer
// }

File: languages/python/exercises_solutions/quick_sort_soln.py
def quick_sort(arr):
  if len(arr) < 2:
    return arr

  pivot = arr[0]
  left = []
  right = []

  for el in arr[1:]:
    if el <= pivot:
      left.append(el)
    else:
      right.append(el)
  
  sorted_left = quick_sort(left)
  sorted_right = quick_sort(right)

  return sorted_left + [pivot] + sorted_right